bin/print_rules.ml

```diff
@@ -60,7 +60,7 @@ let rec encode : Action.For_shell.t -> Dune_lang.t =
   | With_accepted_exit_codes (pred, t) ->
     List
       [ atom "with-accepted-exit-codes"
-      ; Predicate_lang.encode Dune_lang.Encoder.int pred
+      ; Predicate_lang.encode Dune_lang.Encoder.int (fun _ -> assert false) pred
```

Comment:
That looks... odd? Maybe add a comment what this does/why?

Reply:
indeed it's a bit ugly, but I'm not sure what can be done about it. `encode` needs a function to encode the glob only for `string Predicate_lang.t`, but I don't think we can express that with the types.